### PR TITLE
🌱  Release v1.6.3

### PR DESCRIPTION
**Pull request by willie-yao**
/area release

### CODE REVIEW

File: CHANGELOG/v1.6.3.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- CABPK: Add pod metadata to capbk manager (#10212)
+- clusterctl: Add 0 default to worker-machine-count help (#10205)
+- clusterctl: Add hivelocity infra provider to clusterctl (#10168)
+- Dependency: Bump go version to 1.21.8 (#10246)
+- Release: Improve release-staging build (#10190)
+- Release: Read in dockerfiles from stdin (#10209)
+
+:book: Additionally, there have been 3 contributions to our documentation and book. (#10230, #10241, #10243) 
+
+## Dependencies
```
Comment (fabriziopandini) [inline CHANGELOG/v1.6.3.md:27]: Should we drop this or add a "No dependency changes in this release" statement?

**Comment by cahillsf** [comment]
/hold

same as: https://github.com/kubernetes-sigs/cluster-api/pull/10251#issuecomment-1989511632

File: CHANGELOG/v1.6.3.md (hunk 1 above)
Comment (sbueringer) [reply CHANGELOG/v1.6.3.md:28]: Hm. Q: Why is this section empty? We have a bump in l.22

Comment (cahillsf) [reply CHANGELOG/v1.6.3.md:28]: this was updated with the force push just a few minutes ago: https://github.com/kubernetes-sigs/cluster-api/pull/10252/files#r1521783375

Comment (cahillsf) [reply CHANGELOG/v1.6.3.md:28]: does that look right you @sbueringer ?

Comment (sbueringer) [reply CHANGELOG/v1.6.3.md:28]: All good. Sorry I think I missed the sub sections :)

**Pull request merged**